### PR TITLE
Update docs for interleaved and implement integration tests.

### LINKED ISSUE #169: Add docs about Interleaved annotation
- Add docs about the `@Interleaved` annotation.
- Update docs about the delete cascades. Because Interleaved has an on-delete-cascade

### LINKED ISSUE #170: Add Integration Tests for Interleaved feature
Add more integration tests for the interleaved feature.

### PR DESCRIPTION
**Pull request by dzou**
Updates the docs for the Interleaved annotation and also adds integration test to verify. feature.

Fixes #169.
Fixes #170.

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -160,6 +160,49 @@ The following settings are available:
 
 Hibernate performs schema updates on each table and entity type on startup, which can take more than several minutes if there are many tables. To avoid schema updates keeping Hibernate from starting for several minutes, you can update schemas separately and use the `none` or `validate` settings.
 
+==== Interleaved Tables
+
+Cloud Spanner offers the concept of https://cloud.google.com/spanner/docs/schema-and-data-model#creating-interleaved-tables[Interleaved Tables] which allows you to cluster the rows of an interleaved table with rows of a parent table.
```
Comment (meltsufin) [inline README.adoc]:
```suggestion
Cloud Spanner offers the concept of https://cloud.google.com/spanner/docs/schema-and-data-model#creating-interleaved-tables[Interleaved Tables] which allows you to co-locate the rows of an interleaved table with rows of a parent table for efficient retrieval.
```

File: README.adoc [hunk 2]
```diff
@@ -160,6 +160,49 @@ The following settings are available:
...
+This feature enforces the one-to-many relationship and provide efficient queries and operations on entities of a single domain parent entity.
```
Comment (meltsufin) [inline README.adoc]:
```suggestion
This feature enforces the one-to-many relationship and provides efficient queries and operations on entities of a single domain parent entity.
```

File: README.adoc [hunk 3]
```diff
@@ -160,6 +160,49 @@ The following settings are available:
...
+
+If you would like to generate interleaved tables in Spanner, you must annotate your entity with the `@Interleaved` annotation.
```
Comment (meltsufin) [inline README.adoc]:
```suggestion
If you would like to generate interleaved tables in Cloud Spanner, you must annotate your entity with the `@Interleaved` annotation.
```

File: google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/dialect/InterleavedTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+   */
+  @Override
+  protected Map buildSettings() {
+    Map settings = super.buildSettings();
+    settings.put(org.hibernate.cfg.AvailableSettings.HBM2DDL_AUTO, "create-drop");
+    return settings;
+  }
+
+  @Test
+  public void testInterleavedEntities() {
```
Comment (meltsufin) [inline google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/dialect/InterleavedTest.java]: How is this specifically testing `@Interleaved` though? In other words, how would this break if we didn't support interleaved?

Do you need to verify schema generation that it includes the `INTERLEAVED` keyword or something?

Comment (dzou) [reply google-cloud-spanner-hibernate-testing/src/test/java/org/hibernate/dialect/InterleavedTest.java]: Done. Ah good point. I added in the test to now check the database schema to verify that the tables are interleaved with the correct parent.

This was done by querying the `information_schema.tables` table and checking the `parent_table_name` field is set correctly for interleaved tables. See: https://cloud.google.com/spanner/docs/information-schema#tables

**Pull request merged**